be/src/vec/functions/function_string.h

```diff
@@ -5401,8 +5402,137 @@ class FunctionUnicodeNormalize : public IFunction {
 
         output.clear();
         result16.toUTF8String(output);
+=======
+// Implementation of Levenshtein algorithm
+struct LevenshteinImpl {
+    static constexpr auto name = "levenshtein";
+
+    using ResultDataType = DataTypeInt32;
+    using ResultPaddedPODArray = ColumnInt32::Container;
+
+    static DataTypePtr get_return_type_impl(const DataTypes& arguments) {
+        return std::make_shared<ResultDataType>();
+    }
+
+    // Get UTF-8 byte length
+    static size_t get_utf8_byte_length(unsigned char byte) {
+        if (byte < 0x80) return 1;
+        if ((byte & 0xE0) == 0xC0) return 2;
+        if ((byte & 0xF0) == 0xE0) return 3;
+        if ((byte & 0xF8) == 0xF0) return 4;
+        return 1;
+    }
+
+    // Convert string to UTF-8 code points
+    static void to_utf32(const char* data, size_t size, std::vector<int32_t>& out) {
+        out.clear();
+        size_t i = 0;
+        while (i < size) {
+            size_t char_len = get_utf8_byte_length(static_cast<unsigned char>(data[i]));
+            if (i + char_len > size) {
+                char_len = 1;
+            }
+
+            int32_t code_point = 0;
+            for (size_t j = 0; j < char_len; ++j) {
+                code_point = (code_point << 8) | static_cast<unsigned char>(data[i + j]);
+            }
+            out.push_back(code_point);
+            i += char_len;
+        }
+    }
+
+    static void execute(std::string_view l, std::string_view r, int32_t& res) {
+        // Optimization for empty strings
+        if (UNLIKELY(l.empty())) {
+            // Need to count chars in r, not bytes
+            std::vector<int32_t> tmp;
+            to_utf32(r.data(), r.size(), tmp);
+            res = static_cast<int32_t>(tmp.size());
+            return;
+        }
+        if (UNLIKELY(r.empty())) {
+            std::vector<int32_t> tmp;
+            to_utf32(l.data(), l.size(), tmp);
+            res = static_cast<int32_t>(tmp.size());
+            return;
+        }
+
+        std::vector<int32_t> l_code_points;
+        std::vector<int32_t> r_code_points;
+
+        l_code_points.reserve(l.size());
+        r_code_points.reserve(r.size());
+
+        to_utf32(l.data(), l.size(), l_code_points);
+        to_utf32(r.data(), r.size(), r_code_points);
+
+        const auto& source = l_code_points;
+        const auto& target = r_code_points;
+
+        size_t n = source.size();
+        size_t m = target.size();
+
+        // DP arrays: prev_dist (previous row), curr_dist (current row)
+        std::vector<int32_t> prev_dist(m + 1);
+        std::vector<int32_t> curr_dist(m + 1);
+
+        for (size_t j = 0; j <= m; ++j) {
+            prev_dist[j] = static_cast<int32_t>(j);
```

Comment:
we make `j` `size_t` and always cast, it seems weird. you should think clear for why and how we do cast here.